src/display/lv_display.c

```diff
@@ -699,6 +699,25 @@ bool lv_display_delete_event(lv_display_t * disp, uint32_t index)
     return lv_event_remove(&disp->event_list, index);
 }
 
+uint32_t lv_display_remove_event_cb_with_user_data(lv_display_t * disp, lv_event_cb_t event_cb, void * user_data)
+{
+    LV_ASSERT_NULL(disp);
+
+    uint32_t event_cnt = lv_display_get_event_count(disp);
+    uint32_t i;
+    uint32_t removed_count = 0;
+
+    for(i = 0; i < event_cnt; i++) {
```

Comment:
It just came to my mind that it won't work well if there are multiple events to delete after each other. Let's say we want to remove all `event_C_*` from this list:
0. event_A_1
1. event_B_1
2. event_C_1
3. event_C_2
4. event_C_3

When C_1 is deleted with i==2 the list will change like this:
0. event_A_1
1. event_B_1
2. event_C_2
3. event_C_3

In the next iteration `i`  will be `3`, so C_2 is skipped. 

A simple solution would be something like `if(i > 0) i--` after each deletion, or start from the end of the list.

Reply:
Yeah, I think I might have overlooked that.